partition_snapshot_row_cursor.hh

```diff
@@ -111,52 +111,94 @@ class partition_snapshot_row_cursor final {
     friend class partition_snapshot_row_weakref;
     struct position_in_version {
         mutation_partition::rows_type::iterator it;
+        utils::immutable_collection<mutation_partition::rows_type> rows;
```

Comment:
It's odd that position_in_version (an iterator-like thing) has a rows container member. What's the explanation? 

This deserves a comment.

Reply:
It's odd because immutable_collection should be named immutable_view_of_collection. That of course can be done later.

Reply:
> It's odd that position_in_version (an iterator-like thing) has a rows container member. What's the explanation?
> 

I need to compare against rows.begin() during backwards advance(). In forward mode this is not needed since d891cfe6cdd9215613a0ebeeaeb158f4bbc4be5d due to the fact that btree iterators become empty when they reach end()

Reply:
> It's odd because immutable_collection should be named immutable_view_of_collection. That of course can be done later.

Yes, it's a preexisting problem.